pytest_pyvista/pytest_pyvista.py

```diff
@@ -209,10 +224,18 @@ def __call__(self, plotter):  # noqa: ANN001, ANN204, C901, PLR0912
         if self.generated_image_dir is not None:
             gen_image_filename = os.path.join(self.generated_image_dir, test_name[5:] + ".png")  # noqa: PTH118
             plotter.screenshot(gen_image_filename)
+            if not Path(image_filename).is_file():
+                # Image comparison will fail, so save image before error
+                self._save_failed_test_images("error", plotter, image_name)
+
+                # Make sure this doesn't get called again
+                plotter._before_close_callback = None  # noqa: SLF001
```

Comment:
I _think_ this line above was needed because there was an error raised in this code which then causes the plotter to not close properly and leading to this code being called again.  Since there is no error raised here, why is this line here?